sdk/storage/azure-storage-files-shares/test/ut/share_directory_client_test.cpp

```diff
@@ -834,4 +834,157 @@ namespace Azure { namespace Storage { namespace Test {
     {
     }
   }
-}}} // namespace Azure::Storage::Test
+
+  TEST_F(FileShareDirectoryClientTest, AllowTrailingDot)
+  {
+    const std::string directoryName = RandomString();
+    const std::string directoryNameWithTrailingDot = directoryName + ".";
+    const std::string connectionString = StandardStorageConnectionString();
+    const std::string shareName = m_shareName;
+    auto options = InitStorageClientOptions<Files::Shares::ShareClientOptions>();
+
+    auto testTrailingDot = [&](bool allowTrailingDot) {
+      options.AllowTrailingDot = allowTrailingDot;
+
+      auto shareServiceClient = Files::Shares::ShareServiceClient::CreateFromConnectionString(
+          connectionString, options);
+      auto shareClient = shareServiceClient.GetShareClient(shareName);
+      auto rootDirectoryClient = shareClient.GetRootDirectoryClient();
+      auto directoryClient
+          = rootDirectoryClient.GetSubdirectoryClient(directoryNameWithTrailingDot);
+
+      // Create
+      auto createResult = directoryClient.Create().Value;
+
+      // ListFilesAndDirectories
+      auto directories = rootDirectoryClient.ListFilesAndDirectories().Directories;
```

Comment:
This only lists the first page of results. The directory we want to check may not in this page, right?

Reply:
In this case, there are only 2 elements in the list， but considering the integrity of the code, I think your comment is reasonable. I have updated the code to traverse the entire list.